### PR TITLE
Haskell: Correct dependencies for hzk and haskakafka libraries

### PR DESCRIPTION
**Pull request by commandodev**
Depends on (and includes) commits from https://github.com/NixOS/nixpkgs/pull/6121

### CODE REVIEW

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -54823,7 +54823,7 @@ self: {
        description = "Kafka bindings for Haskell";
        license = stdenv.lib.licenses.mit;
        hydraPlatforms = stdenv.lib.platforms.none;
-     }) { rdkafka = null;};
```
Comment (peti) [inline pkgs/development/haskell-modules/hackage-packages.nix]: Please don't edit this file manually! See the comment in the first line; changes you make here will be lost the next time this file is re-generated.

**Comment by peti** [comment]
Merged in aa39df3 and b1424af.

**Comment by boothead** [comment]
I was assuming that these we're the correct edits to simulate what would happen next time the file was re-generated. For future reference, what's the best workflow to use here?

 * Should I run the regeneration tool myself?
 * Make the changes locally but not include in the commit?
 * Change hackage-packages.nix to simulate the edits I made and then back them out after the file is regenerated?

What work flow do you guys use?

BTW, I'm really liking the new haskellng stuff, great work!
Cheers,
Ben

**Comment by peti** [comment]
Now that I think about it, you are right: there's no good solution other than editing the `hackage-packages.nix` file. You need the `= null` overrides gone to make your packages build.

I guess, one could run `hackage2nix` to re-generate the file from scratch. The downside is that this would probably introduce some package updates as a side-effect that you don't want.

Interesting problem ... there's no good alternative to manually editing the file. The danger I see with the manual edits is that the re-generation process may come up with some output that isn't what you expected, and then the builds might be broken without you noticing.

**Pull request closed without merging**